extensions/positron-python/src/client/application/diagnostics/checks/unsupportedPythonVersion.ts

```diff
@@ -0,0 +1,141 @@
+            this.timeOut = undefined;
+        }
+    }
+
+    public async diagnose(resource: Resource): Promise<IDiagnostic[]> {
+        const interpreterService = this.serviceContainer.get<IInterpreterService>(IInterpreterService);
+        const interpreter = await interpreterService.getActiveInterpreter(resource);
+        if (isVersionSupported(interpreter?.version)) {
+            return [];
+        }
+        return [new UnsupportedPythonVersionDiagnostic(DiagnosticCodes.UnsupportedPythonVersion, resource)];
+    }
+
+    protected async onHandle(diagnostics: IDiagnostic[]): Promise<void> {
+        if (diagnostics.length === 0) {
+            return;
+        }
+        const messageService = this.serviceContainer.get<IDiagnosticHandlerService<MessageCommandPrompt>>(
+            IDiagnosticHandlerService,
+            DiagnosticCommandPromptHandlerServiceId,
+        );
+        await Promise.all(
+            diagnostics.map(async (diagnostic) => {
+                const canHandle = await this.canHandle(diagnostic);
+                const shouldIgnore = await this.filterService.shouldIgnoreDiagnostic(diagnostic.code);
+                if (!canHandle || shouldIgnore) {
+                    return;
+                }
+                const commandPrompts = this.getCommandPrompts(diagnostic);
+                await messageService.handle(diagnostic, { commandPrompts, message: diagnostic.message });
+            }),
+        );
+    }
+
+    protected addPythonPathChangedHandler(): void {
+        const disposables = this.serviceContainer.get<IDisposableRegistry>(IDisposableRegistry);
+        const interpreterPathService = this.serviceContainer.get<IInterpreterPathService>(IInterpreterPathService);
+        disposables.push(interpreterPathService.onDidChange((i) => this.onDidChangeConfiguration(i)));
+    }
+
+    protected async onDidChangeConfiguration(
+        interpreterConfigurationScope: InterpreterConfigurationScope,
+    ): Promise<void> {
+        const workspaceUri = interpreterConfigurationScope.uri;
+        if (this.timeOut && typeof this.timeOut !== 'number') {
+            clearTimeout(this.timeOut);
+            this.timeOut = undefined;
+        }
+        this.timeOut = setTimeout(() => {
+            this.timeOut = undefined;
+            this.diagnose(workspaceUri)
+                .then((diagnostics) => this.handle(diagnostics))
+                .ignoreErrors();
+        }, this.changeThrottleTimeout);
+    }
+
+    private getCommandPrompts(diagnostic: IDiagnostic): { prompt: string; command?: IDiagnosticCommand }[] {
+        const commandFactory = this.serviceContainer.get<IDiagnosticsCommandFactory>(IDiagnosticsCommandFactory);
+        switch (diagnostic.code) {
+            case DiagnosticCodes.UnsupportedPythonVersion: {
+                return [
+                    {
+                        prompt: Common.selectPythonInterpreter,
+                        command: commandFactory.createCommand(diagnostic, {
```

Comment:
we probably want this to go to the Positron set interpreter rather than the vscode one